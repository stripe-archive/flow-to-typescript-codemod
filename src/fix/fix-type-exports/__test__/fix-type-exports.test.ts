import { fixTypeExports } from '..'
import {
  expectMigrationReporterMethodCalled,
  createOutputRecorder,
  getTestFixState,
} from '../../../convert/utils/testing'

jest.mock('../../../runner/migration-reporter')
jest.mock('../../../runner/logger')

describe('fixTypeExports', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('fixes mismatched type imports and exports', async () => {
    const [results, recordTestResult] = createOutputRecorder()

    await fixTypeExports(
      getTestFixState({
        tsProps: false,
        autoSuppressErrors: false,
        generateReport: false,
        jiraSlug: '',
        useIgnore: false,
        removeUnused: false,
        config: './src/fix/fix-type-exports/__test__/tsconfig.json',
        format: 'stdout',
        silent: false,
        output: '',
        autoImport: false,
        fixTypeExports: false,
      }),
      recordTestResult
    )

    expectMigrationReporterMethodCalled(`typeExports`)

    expect(results['test-input.ts']).toMatchSnapshot()
  })
})
