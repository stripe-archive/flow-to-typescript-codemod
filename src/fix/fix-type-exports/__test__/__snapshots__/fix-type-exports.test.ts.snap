// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixTypeExports fixes mismatched type imports and exports 1`] = `
"import React, { Component, ElementType, RefObject } from 'react'

// Comment #1

export type { Key, ReactNode as Node } from 'react'

// Comment #2

export { Component }

// Comment #3
export type { ElementType, RefObject };
"
`;
